tests/Libraries/BeatmapsetDiscussion/DiscussionTest.php

```diff
@@ -0,0 +1,420 @@
+        $this->expectCountChange(fn () => BeatmapDiscussionPost::count(), $change, BeatmapDiscussionPost::class);
+        $this->expectCountChange(fn () => Notification::count(), $change, Notification::class);
+        $this->expectCountChange(fn () => UserNotification::count(), $change, UserNotification::class);
+
+        if ($verified) {
+            $user->markSessionVerified();
+        }
+
+        if (!$success) {
+            $this->expectException(VerificationRequiredException::class);
+        }
+
+        (new Discussion($user, $beatmapset, $this->makeParams('praise'), static::TEST_MESSAGE))->handle();
+    }
+
+    /**
+     * See testReopeningProblemDoesNotDisqualifyOrResetNominations for assertions
+     * jobs are not queued when reopening a resolved discussion.
+     *
+     * @dataProvider newDiscussionQueuesJobsDataProvider
+     */
+    public function testNewDiscussionQueuesJobs(string $state, ?string $group, array $queued, array $notQueued)
+    {
+        $user = User::factory()->withGroup($group)->create()->markSessionVerified();
+
+        $beatmapset = $this->beatmapsetFactory()
+            ->withNominations()
+            ->$state()
+            ->create();
+
+        Queue::fake();
+
+        (new Discussion($user, $beatmapset, $this->makeParams('problem'), static::TEST_MESSAGE))->handle();
+
+        foreach ($queued as $class) {
+            Queue::assertPushed($class);
+        }
+
+        foreach ($notQueued as $class) {
+            Queue::assertNotPushed($class);
+        }
+    }
+
+    /**
+     * @dataProvider shouldDisqualifyOrResetNominationsDataProvider
+     */
+    public function testShouldDisqualifyOrResetNominations(string $state, ?string $group, string $messageType, bool $expects)
+    {
+        $user = User::factory()->withGroup($group)->create()->markSessionVerified();
+
+        $beatmapset = $this->beatmapsetFactory()
+            ->withNominations()
+            ->$state()
+            ->create();
+
+        $subject = new Discussion($user, $beatmapset, $this->makeParams($messageType), static::TEST_MESSAGE);
+
+        $value = $this->invokeMethod($subject, 'shouldDisqualifyOrResetNominations');
+        $this->assertSame($expects, $value);
+    }
+
+    public function testWatchersGetNotification()
+    {
+        $user = User::factory()->create()->markSessionVerified();
+        $watcher = User::factory()->create();
+        $beatmapset = $this->beatmapsetFactory()->create();
+        $beatmapset->watches()->create(['user_id' => $watcher->getKey()]);
+
+        Queue::fake();
+
+        (new Discussion($user, $beatmapset, $this->makeParams('praise'), static::TEST_MESSAGE))->handle();
+
+        Queue::assertPushed(BeatmapsetDiscussionPostNew::class, function (BeatmapsetDiscussionPostNew $job) use ($user, $watcher) {
+            return in_array($watcher->getKey(), $job->getReceiverIds(), true)
+                && !in_array($user->getKey(), $job->getReceiverIds(), true);
+        });
```

Comment:
kinda long line. Maybe arguments on their own line and use the short anonymous function syntax

also similarly for some others